packages/eslint-plugin-js/rules/comma-style/comma-style.ts

```diff
@@ -166,66 +172,65 @@ export default {
 
     /**
      * Checks the comma placement with regards to a declaration/property/element
-     * @param {ASTNode} node The binary expression node to check
-     * @param {string} property The property of the node containing child nodes.
+     * @param node The binary expression node to check
+     * @param property The property of the node containing child nodes.
      * @private
-     * @returns {void}
      */
-    function validateComma(node, property) {
-      const items = node[property]
+    function validateComma<T extends NodeType, const K extends keyof T>(node: T, property: K): void {
+      const items = node[property] as (TSESTree.Token | TSESTree.Node)[]
       const arrayLiteral = (node.type === 'ArrayExpression' || node.type === 'ArrayPattern')
 
       if (items.length > 1 || arrayLiteral) {
         // seed as opening [
-        let previousItemToken = sourceCode.getFirstToken(node)
+        let previousItemToken = sourceCode.getFirstToken(node)!
 
         items.forEach((item) => {
-          const commaToken = item ? sourceCode.getTokenBefore(item) : previousItemToken
-          const currentItemToken = item ? sourceCode.getFirstToken(item) : sourceCode.getTokenAfter(commaToken)
-          const reportItem = item || currentItemToken
+          const commaToken = item ? sourceCode.getTokenBefore(item)! : previousItemToken
+          const currentItemToken = item ? sourceCode.getFirstToken(item as TSESTree.Node)! : sourceCode.getTokenAfter(commaToken)!
+          const reportItem = item ?? currentItemToken
 
           /*
```

Comment:
```suggestion
          /**
```

This would allow ESLint to auto fix it